datasync/chaindatafetcher/kas/repository.go

```diff
@@ -98,6 +100,21 @@ func (r *repository) SetComponent(component interface{}) {
 	}
 }
 
+func (r *repository) HandleChainEvent(event blockchain.ChainEvent, reqType types.RequestType) error {
+	switch reqType {
+	case types.RequestTypeTransaction:
+		return r.InsertTransactions(event)
+	case types.RequestTypeTrace:
+		return r.InsertTraceResults(event)
+	case types.RequestTypeTokenTransfer:
+		return r.InsertTokenTransfers(event)
+	case types.RequestTypeContract:
+		return r.InsertContracts(event)
+	default:
+		return fmt.Errorf("unsupported data type: %v", reqType)
```

Comment:
How about also leaving the information from ChainEvent?

Reply:
Added block number to the error. 1c477b9014671902ffff40079ea39a2142812750